pkg/internal/kube/store.go

```diff
@@ -339,16 +339,21 @@ func (s *Store) ServiceNameNamespaceForIP(ip string) (string, string) {
 		s.access.RUnlock()
 		return serviceInfo.Name, serviceInfo.Namespace
 	}
+	s.access.RUnlock()
+
+	s.access.Lock()
+	defer s.access.Unlock()
+
+	if serviceInfo, ok := s.otelServiceInfoByIP[ip]; ok {
+		return serviceInfo.Name, serviceInfo.Namespace
+	}
```

Comment:
NIT: This operation is read-only. Why do you wrap into the mutex Write lock?

Reply:
I think you are right, we don't need that one under the lock, the read after was important to be inside the write lock.